src/storage/src/hummock/cache.rs

```diff
@@ -677,6 +680,37 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
     }
 }
 
+impl<K: LruKey + Clone, T: LruValue> LruCache<K, T> {
+    pub async fn lookup_with_request_dedup<F, VC>(
+        self: &Arc<Self>,
+        hash: u64,
+        key: K,
+        fetch_value: F,
+    ) -> HummockResult<CachableEntry<K, T>>
+    where
+        F: FnOnce() -> VC,
+        VC: Future<Output = HummockResult<(T, usize)>>,
+    {
+        match self.lookup_for_request(hash, key.clone()) {
+            LookupResult::Cached(entry) => Ok(entry),
+            LookupResult::WaitPendingRequest(recv) => {
+                let entry = recv.await.map_err(HummockError::other)?;
```

Comment:
Better to return the original error to the caller, instead of `channel closed` or something unclear. Here we can only get channel closed error.

Reply:
Previously, `moka` will return `Arc<Error>` to all callers.

Reply:
The original error may not support `Clone`?